### ISSUE TITLE
Niantic detect the Bot

### ISSUE THREAD

**Issue by mariuserik** [PokemonGoF/PokemonGo-Bot#3722]
Currently using dev branch bot.

First minutes using the bot work (5 minutes), I've tested with different connections and accounts (work and stop in few minutes) the bot still runs but without finding any pokemon.

It means Niantic knows exactly who is using the bot.

![bildschirmfoto 2016-08-12 um 12 34 11](https://cloud.githubusercontent.com/assets/1163058/17619926/e5613436-6088-11e6-84a0-379960e0db67.png)

**Comment by takenek** [comment]
Yep exacly i saw this same. After morning git pull my bot was banned... i Create new server with new IP and login/password after 10 minutes i get ban.

**Comment by fari-99** [comment]
if you are using more than 2 bots, you will get IP softban less than 30 min. it will wore of after 1 hour or so. i use the 1 bot, and it run 7 hours now.

**Comment by z4ppy** [comment]
fari-99: are you using 3G/4G connexion ?

**Comment by nivong** [comment]
https://github.com/PokemonGoF/PokemonGo-Bot/issues/3720
@k4n30

**Comment by mariuserik** [comment]
I tested, new connection using only one bot, I have the same problem, banned after a few minutes.

**Comment by takenek** [comment]
i use new login/password, new installed server and i use new IP address and after 10 minutes get ban...
So they find a way to detect a bots...

**Comment by divrot** [comment]
game over?

**Comment by divrot** [comment]
Takenek, what was your move speed?

**Comment by MeowoeM** [comment]
It seems that Niantic should be detecting if your connection is 3G/4G. After I connect my laptop to the hot spot of my cell phone, the bot works again.

**Comment by divrot** [comment]
> After I connect my laptop to the hot spot of my cell phone, the bot works again.
Try connect phone to wi-fi than PC to phone

**Comment by fari-99** [comment]
@MeowoeM thats not right. if someone in cafe and play pokemon with wifi then they get banned? i don't think so.

@z4ppy i'm using home wifi

**Comment by reconka** [comment]
its a simple  ip ban..   dont worry

**Comment by wubbel** [comment]
I switched on my phone to 3G connection but still there are no pokemon or stops on the map.

**Comment by YuMan-Tam** [comment]
Exactly the same situation. It works the whole past night until about an hour ago.

**Comment by buzzo123** [comment]
tes, same ban with 2 bot, now i try with one

**Comment by tonyschuite** [comment]
4G works for me... both on dev as on master / Desktop

**Comment by MeowoeM** [comment]
@fari-99 @divrot 
IP ban is more probable. As you said, when i reconnect my phone to wifi, nothing appears in pg. My last guessing is misleading. sorry for that

**Comment by mariuserik** [comment]
@fari-99.  You don't understand the problem? this is a real issue, you should not close a not solved issue.

**Comment by fari-99** [comment]
@MeowoeM no problem.

@mariuserik the developer **need more information**!! if you just give the dev like followspiral bug, can't spinfort and etc. you can't work with it!! there is already issue like this everywhere in this repo, maybe if you kind enough to search it, maybe your issue get solved!!

**Comment by PokeProgress** [comment]
Woah i think niantic figured out how to terminate the bots

**Comment by mariuserik** [comment]
@fari-99 I say exactly what is happening. Niantic detects that you are using bots. I've opened a Issue so that we can discuss it. Thats how this "Issue" page from github works like. Its easily reproduced, so that developers may debug it.

**Comment by ctsstc** [comment]
Might have been coincidence but this happened to me after doing a "git pull" wish I knew which SHA I was on before doing so. I think everyone started to get hit about the same time as when I did this (so we can just all blame me xP) 
According to chat some guy was running many other bots ~1600, around 40 of which were this bot and they had around a 75%+ ban rate, while his other bots sound to be un-touched.¿ Time will tell...

**Comment by z4ppy** [comment]
@mariserik: You do not understand, niantec does not detect robots but the type of connection you use. I think for now they only accept connections from that IP telephone operator! (Some ISPs still work elsewhere;)

**Comment by fari-99** [comment]
@mariuserik yes, use that issue page for discussing, please. not create another issue. thats why we **want** you to see the other issue and begin discussion there. btw that's why we use search engine.

**Comment by buzzo123** [comment]
when u run at least 2 bot, the bot just walk , but when u stop it walk attribute say 0 km, you have to reset the modem and if u play with only one both work fine

**Comment by YuMan-Tam** [comment]
I am not sure, but this could be a different issue. The references above suggested that the problem could be solved by manually changing one's location. I just did - nothing changed.

**Comment by PokeProgress** [comment]
i failed to get game data from the server on one of my accounts

**Comment by fari-99** [comment]
@PokeProgress maybe this will help #3694

**Comment by k4n30** [comment]
#3308 #3394 #3475 #3646 #3692 #3700 #3711 #3721 #3722

**Comment by mariuserik** [comment]
I would recommend to anyone not to use the bot, till this is fixed. There are cases of permanent ban. And a few discussions are running on Reddit

https://www.reddit.com/r/pokemongodev/comments/4xct8f/psa_permabans_have_arrived_stop_all_use_of_3rd/

https://np.reddit.com/r/TheSilphRoad/comments/4xcvl1/bots_and_spoofers_receiving_permanent_bans/

**Comment by slickochet** [comment]
Same problem today after I did a Git pull.
But I reverted back to an older version from last night (11/08/2016) and it works again.  So some recent commit seems to be causing an issue.

I didn't run it for very long however because of the rumours of permabans going around now until there is more information...

**Comment by durd** [comment]
```
2016-08-12 13:24:53,011 [pgoapi.rpc_api] [DEBUG] Protobuf structure of rpc response:
status_code: 3
request_id: 88156449872972132
returns: ""

2016-08-12 13:24:53,067 [pgoapi.rpc_api] [DEBUG] Decode raw over protoc (protoc has to be in your PATH):
Couldn't find protoc in your environment OR other issue...
```

**Comment by mariuserik** [comment]
@k4n30. You just closed alot of duplicate. But dont even say which Issue is the "original", in which ticket is this Issue being discussed?

**Comment by k4n30** [comment]
@mariuserik trying to set that up now

**Comment by k4n30** [comment]
#3682 - META Server busy or offline issues
#3308 - META Stuck between two forts / Moving to 2 forts at the same time / Constant Loop
#3700 - META Soft bans - Try again spins / Pokemon vanishing

**Comment by priyadi** [comment]
I got the same problem after git pulling today. Checking out d7cb7df27516b588252a767260176f92586489fa fixes the problem for me.

**Comment by avexus** [comment]
from what i read from this, it is NOT safe to use this bot any more. Am I right?
